src/FileLoader.php

```diff
@@ -0,0 +1,208 @@
+
+            }else{
+                $this->unsupportedFiles[] = $file;
+            }
+        }
+
+        if ( ! empty($this->unsupportedFiles)){
+            $badFiles = implode(", ", $this->unsupportedFiles);
+            throw new UnsupportedFilesException('The file(s) '. $badFiles .' are not supported by the available decoders.');
+        }
+
+        if ($append){
+            $this->addDataToManager();
+            return;
+        }
+        $this->resetManagerAndAddData();
+    }
+
+    /**
+     * Check to make sure the mime type is ok.
+     * @param $type a mime type
+     * @return bool
+     */
+    protected function isSupportedMimeType($type)
+    {
+        return in_array($type, $this->supportedMimeTypes);
+    }
+
+    /**
+     * Resets the file loader back to initial state.
+     */
+    public function reset()
+    {
+        $this->fileBag = array();
+        $this->supportedMimeTypes = array();
+        $this->unsupportedFiles = array();
+        $this->dataToDecode = array();
+    }
+
+    /**
+     * Returns the contents of the file.
+     *
+     * @return string the contents of the file
+     *
+     * @throws \RuntimeException
+     */
+    protected function getFileContents($file)
+    {
+        if ( $file->getExtension() === 'php') {
+            $content = include $file->getPathname();
+            return $content;
+        }
+
+        $level = error_reporting(0);
+        $content = file_get_contents($file->getPathname());
+        error_reporting($level);
+        if (false === $content) {
+            $error = error_get_last();
+            throw new \RuntimeException($error['message']);
+        }
+        return $content;
+    }
+
+    /**
+     * This method appends data to an existing manager.
+     */
+    protected function addDataToManger()
+    {
+        foreach ($this->decodedData as $data)
+        {
+            $this->manager->add($data);
+        }
+    }
+
+    /**
+     * This method will start with a fresh manager and then add data to it.
+     */
+    protected function resetManagerAndAddData()
+    {
+        $reset = true;
+        foreach ($this->decodedData as $data)
+        {
+            if ($reset === true){
+                $this->manager->reset($data);
+                $reset = false;
+            }
+            $this->manager->add($data);
+        }
+    }
+
+    /**
+     * Default decoder class factory method.
+     * Checks to make sure we have a default decoder available and if so, adds it as a decoder to the file loader.
+     *
+     * @param $mimeType
+     */
+    protected function checkAndAddDefaultDecoder($mimeType)
+    {
+        $decoderClass = ucfirst($mimeType)."Decoder";
+
+        if(file_exists(__DIR__.'/Decoders/'.$decoderClass.'.php') && ! $this->isSupportedMimeType($mimeType)){
+
+            include_once(__DIR__.'/Decoders/'.$decoderClass.'.php');
+            $nameSpace = "Michaels\\Manager\\Decoders\\";
+            $fqcn = $nameSpace . $decoderClass;
```

Comment:
With IDE's, its simple enough to use more complete variable names. Maybe even `fullClass`.

Also, we use Composer to autoload files, so we don't need to include that file manually. Just remove 200